core/clustersmngr/factory.go

```diff
@@ -314,10 +327,27 @@ func (cf *clustersManager) watchNamespaces(ctx context.Context) {
 	}
 }
 
+// Used when all clusters have kubeconfig secrets that contain credentials to connect to the cluster
+// and list their namespaces.
 func (cf *clustersManager) UpdateNamespaces(ctx context.Context) error {
+	return cf.updateNamespacesWithClient(ctx, func() (Client, error) {
+		return cf.GetServerClient(ctx)
+	})
+}
+
+// Used when clusters have kubeconfig secrets that DO NOT contain credentials to connect to the cluster.
+// We need to use the user's token from the incoming request to connect to the cluster and list its namespaces
+// instead.
+func (cf *clustersManager) updateNamespacesFromUserClient(ctx context.Context, user *auth.UserPrincipal) error {
+	return cf.updateNamespacesWithClient(ctx, func() (Client, error) {
+		return cf.getImpersonatedClientWithNamespaces(ctx, user, false)
```

Comment:
maybe this is a bit confusing, it's explicitly _not_ using impersonation?

Reply:
True!